onnxruntime/core/graph/graph.cc

```diff
@@ -4071,8 +4071,11 @@ Status Graph::InlineFunctionProto(const ONNX_NAMESPACE::FunctionProto& func_to_i
     InlinedVector<onnxruntime::NodeArg*> inputs;
     InlinedVector<onnxruntime::NodeArg*> outputs;
 
-    for (const auto& tensor_name : inlined_node->input())
-      inputs.push_back(to_node_arg(tensor_name));
+    for (const auto& tensor_name : inlined_node->input()) {
+      if (!tensor_name.empty()) {
```

Comment:
I don't understand this. The empty name input is primarily meant for missing optional inputs that are followed by other (nonmissing) inputs ... that is, for non-trailing missing optional inputs. How exactly does onnxruntime handle them? My memory (this may be outdated) is that onnxruntime does something similar, allowing empty-named nodeargs.

I am concerned that this logic here will lose the information about the positional-index of the node-args.

Reply:
I suggest that this be restricted to the "trailing" case, where there are no non-empty input following this one.